src/pip/_internal/operations/check.py

```diff
@@ -77,7 +76,7 @@ def should_ignore(name):
         missing_deps = set()  # type: Set[Missing]
         conflicting_deps = set()  # type: Set[Conflicting]
 
-        if should_ignore(package_name):
+        if should_ignore_package(package_name):
```

Comment:
```suggestion
        if should_ignore and should_ignore(package_name):
```

This would be simpler, IMO.

Reply:
Agreed, that makes it simpler. Added in https://github.com/pypa/pip/pull/8322/commits/1da909ecd62947504324c3f015d869ce3fea0fe2